### PR TITLE
Update AutoClone.md

### PR DESCRIPTION
**Pull request by oasaleh**
This pull request includes updates to the documentation to clarify the behavior of the `GetOrDefault` method when using the `SetAutoClone` option in the `FusionCache` library.

Documentation improvements:

* [`docs/AutoClone.md`](diffhunk://#diff-52745695719e913d6501b409cc8f544ec9b0106f18ffd96a8cb02973ea151a43L85-R106): Updated comments to clarify that using `SetAutoClone(true)` returns a clone of the cached instance, and modifications affect only the clone, while not using `SetAutoClone` returns a direct reference to the cached instance, and modifications affect the cache.

### CODE REVIEW

**Review by jodydonetti — APPROVED** [review]
Sorry for the delay.
I liked your changes, they made it more clear. I also made a couple of minor adjustements.

LGTM, what do you think?

File: docs/AutoClone.md [hunk 1]
```diff
@@ -82,28 +82,28 @@ cache.SetupSerializer(new FusionCacheSystemTextJsonSerializer());
...
 cache.Set("foo", new Person { Name = "John" });
 
-// RETURNS A CLONE OF CACHED INSTANCE (ORIGINAL REMAINS UNCHANGED)
+// RETURNS A CLONE OF THE CACHED INSTANCE
 var person1 = cache.GetOrDefault<Person>("foo", options => options.SetAutoClone(true));
 Console.WriteLine($"person1: {person1.Name}");
 Console.WriteLine();
 
-// RETURNS A CLONE OF CACHED INSTANCE (MODIFICATIONS AFFECT ONLY THE CLONE)
+// RETURNS A CLONE OF THE CACHED INSTANCE: CHANGES APPLIED ONLY THE CLONE, CACHED INSTANCE REMAINS UNCHANGED
```
Comment (oasaleh) [inline docs/AutoClone.md:90]: I noticed a couple of points regarding this comment:

1. The phrase “applied only” should be updated to “applied only to” for grammatical correctness.
2. The comment feels slightly verbose, which makes it less clear. For example, the term “clone” is used without pairing it with “instance”. Using “cloned instance” would improve clarity and consistency. I think reverting it back to `(MODIFICATIONS AFFECT ONLY THE CLONE)` would be ideal. However, this version could also work: 
`// RETURNS A CLONED INSTANCE OF THE CACHED OBJECT: CHANGES ARE APPLIED ONLY TO THE CLONED INSTANCE, LEAVING THE CACHED INSTANCE UNMODIFIED`

File: docs/AutoClone.md [hunk 2]
```diff
@@ -82,28 +82,28 @@ cache.SetupSerializer(new FusionCacheSystemTextJsonSerializer());
...
 var person2 = cache.GetOrDefault<Person>("foo", options => options.SetAutoClone(true));
 person2.Name = "Jane";
 Console.WriteLine($"person1: {person1.Name}");
 Console.WriteLine($"person2: {person2.Name}");
 Console.WriteLine();
 
-// RETURNS DIRECT REFERENCE TO CACHED INSTANCE (MODIFICATIONS AFFECT THE CACHE)
+// RETURNS DIRECT REFERENCE TO THE CACHED INSTANCE: CHANGES APPLIED TO THE CACHED INSTANCE ITSELF
```
Comment (oasaleh) [inline docs/AutoClone.md:97]: Same as the comment above. Also, it's missing the verb "are" in "Changes _are_ applied to", which makes it verbose. 🤷

**Review by jodydonetti — APPROVED** [review]
LGTM

**Pull request merged**